crates/refunder/src/refund_service.rs

```diff
@@ -1,46 +1,101 @@
 use anyhow::Result;
-use database::ethflow_orders::refundable_orders;
+use contracts::CoWSwapEthFlow;
+use database::{
+    ethflow_orders::{mark_eth_orders_as_refunded, refundable_orders},
+    OrderUid,
+};
+use lazy_static::lazy_static;
+use primitive_types::H160;
+use shared::{transport::MAX_BATCH_SIZE, Web3, Web3CallBatch};
 use sqlx::PgPool;
 use std::time::SystemTime;
 
+// The following address is used in the ethflow contract to mark invalidated orders
+lazy_static! {
+    static ref INVALIDATED_OWNER: H160 = H160([255u8; 20]);
+}
+
 pub struct RefundService {
     pub db: PgPool,
+    pub web3: Web3,
+    pub ethflow_contract: CoWSwapEthFlow,
     pub min_validity_duration: i64,
     pub min_slippage: f64,
 }
 
+type SplittedOrderUids = (Vec<(OrderUid, bool)>, Vec<(OrderUid, bool)>);
+
 impl RefundService {
-    pub fn new(db: PgPool, min_validity_duration: i64, min_slippage_in_bps: u64) -> Self {
+    pub fn new(
+        db: PgPool,
+        web3: Web3,
+        ethflow_contract: CoWSwapEthFlow,
+        min_validity_duration: i64,
+        min_slippage_bps: u64,
+    ) -> Self {
         RefundService {
             db,
+            web3,
+            ethflow_contract,
             min_validity_duration,
-            min_slippage: min_slippage_in_bps as f64 / 10000f64,
+            min_slippage: min_slippage_bps as f64 / 10000f64,
         }
     }
 
     pub async fn try_to_refund_all_eligble_orders(&self) -> Result<()> {
-        // Step 1:
         // Look for all refundable orders by calling the database
         let unix_timestamp = SystemTime::now()
             .duration_since(SystemTime::UNIX_EPOCH)
             .unwrap()
             .as_secs() as i64;
         let mut ex = self.db.acquire().await?;
-        let _refundable_order_uids = refundable_orders(
+        let refundable_order_uids = refundable_orders(
             &mut ex,
             unix_timestamp,
             self.min_validity_duration,
             self.min_slippage,
         )
         .await?;
 
-        // Step 2:
-        // take out orderUids that are already refunded
-        // by making a batched web3 call. Update refunded ones.
+        // Identify orderUids that are already refunded
+        // by making a batched web3 call.
```

Comment:
Instead of having a single long function with comments in between you could also have multiple smaller well named functions to make it a bit easier to read in the end.